packages/lisk-transaction-pool/src/transaction_pool.ts

```diff
@@ -311,48 +343,72 @@ export class TransactionPool {
 		);
 	}
 
-	private async processVerifiedTransactions(): Promise<CheckTransactionsResponse> {
+	private async processVerifiedTransactions(): Promise<
+		CheckTransactionsResponseWithPassAndFail
+	> {
 		const transactionsInReadyQueue = this._queues.ready.size();
 		const transactionsInVerifiedQueue = this._queues.verified.size();
+		const processableTransactionsInPendingQueue = this._queues.pending.sizeBy(
+			transaction => transaction.isReady(),
+		);
 
 		if (
-			transactionsInReadyQueue >= this._verifiedTransactionsProcessingLimitPerInterval ||
-			transactionsInVerifiedQueue === 0
+			transactionsInReadyQueue >=
+				this._verifiedTransactionsProcessingLimitPerInterval ||
+			(transactionsInVerifiedQueue === 0 &&
+				processableTransactionsInPendingQueue === 0)
 		) {
 			return {
 				passedTransactions: [],
-				failedTransactions: []
+				failedTransactions: [],
 			};
-		} 
+		}
+
+		const additionalTransactionsToProcessLimit =
+			this._verifiedTransactionsProcessingLimitPerInterval -
+			transactionsInReadyQueue;
+		const transactionsFromPendingQueueLimit = Math.min(
+			additionalTransactionsToProcessLimit,
+			this._pendingTransactionsProcessingLimit,
+		);
+		// Filter at max transactionsFromPendingQueueLimit from the pending queue which are also ready
+		const transactionsFromPendingQueue = this._queues.pending
+			.filter(transaction => transaction.isReady())
```

Comment:
shouldn't we create `peekUntil` for this one too?

Reply:
Ready transactions in the pending queue are placed in an unordered fashion. Therefore, we can't use `peekUntil` function.